packages/core/src/browser/dialogs.ts

```diff
@@ -148,15 +148,30 @@ export abstract class AbstractDialog<T> extends BaseWidget {
         this.addKeyListener(document.body, Key.ENTER, e => this.handleEnter(e));
     }
 
+    protected canHandle(event: KeyboardEvent): boolean {
+        let result: boolean = false;
+        if (this.id && event.currentTarget) {
```

Comment:
I'm not sure checking `this.id` is necessary as `id` will always be defined by the parent `AbstractDialog<T>`:

https://github.com/theia-ide/theia/blob/cc51575a2c8660f13794ed9099ee14daaca982b9/packages/core/src/browser/dialogs.ts#L76

Reply:
@vince-fugnitto I think it's necessary.

Reply:
Can you clarify why it's necessary? (`this.id` is always `true`)
If no `id` is provided by a child, the parent's `theia-dialog-shell` will be used due to `super(props)`.

Reply:
@vince-fugnitto 
I have try add 'event.preventDefault();event.stopPropagation();' into handle func，but it does't work，to binding 'document.body' maybe too wide.

Reply:
I am thinking of a way to solve the problem you mentioned.

Reply:
this.id = `theia-dialog-${(new Date()).getTime()}`;

Reply:
@vince-fugnitto @akosyakov 
In principle, each dialog should have a unique ID, similar to the processing of widgets in the application-shell
The current processing logic uses default timestamps to identify uniqueness.
I think this ID does not allow external changes to avoid exceptions caused by resets.

Reply:
> I have try add 'event.preventDefault();event.stopPropagation();' into handle func，but it does not work，to binding 'document.body' maybe too wide.

Does it stop propagation in wrong order?

Reply:
@akosyakov yes